napari/components/dims.py

```diff
@@ -223,6 +227,36 @@ def set_range(
                     ax = assert_axis_in_bounds(int(ax), self.ndim)
                     full_range[ax] = r
                 self.range = full_range
+        if restore_point is not None:
+            new_point = self._get_point_in_ndim(restore_point)
+            self.set_point(range(self.ndim), new_point)
+
+    def _get_point_in_ndim(
+        self,
+        point: Sequence[float],
```

Comment:
```suggestion
        point: Tuple[float],
```
This code will work only with tuple